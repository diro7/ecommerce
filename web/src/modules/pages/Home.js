// Imports
import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'
import { Link } from 'react-router-dom'

// UI Imports
import { Grid, GridCell } from '../../ui/grid'
import { H1, H4 } from '../../ui/typography'
import Button from '../../ui/button'
import { white } from '../../ui/common/colors'
import { textLevel1 } from '../../ui/common/shadows'

// App Imports
import { APP_URL } from '../../setup/config/env'
import crateRoutes from '../../setup/routes/crate'
import userRoutes from '../../setup/routes/user'
import Onboarding from './Onboarding'

// Component
const Home = (props) => (
  <div>
    {/* Home */}
    <Grid alignCenter={true} style={{
      backgroundImage: `url('${ APP_URL }/images/cover.jpg')`,
      backgroundAttachment: 'fixed',
      backgroundSize: 'cover',
      backgroundPosition: 'center top',
      height: 'calc(100vh - 5em)',
      textAlign: 'center',
      color: white
    }}>
      {/* SEO */}
      <Helmet>
        <title>Suministro mensual de ropa y accesorios para hombres y mujeres - Canasta</title>
      </Helmet>

      {/* Content */}
      <GridCell>
        <H1 font="secondary" style={{ textShadow: textLevel1 }}>Crate</H1>

        <H4 style={{ textShadow: textLevel1, marginTop: '0.5em' }}>
          Suscripción mensual de ropa y accesorios de moda
        </H4>

        {/* Call to action */}
        {
          
            <Link to={userRoutes.signup.path}>
                <Button theme="secondary" style={{ marginTop: '1em' }}>Comience aquí</Button>
              </Link>
        }
      </GridCell>
    </Grid>

    {/* Onboarding */}
    <Onboarding/>
  </div>
)
// Component State
function homeState(state) {
  return {
    user: state.user
  }
}

export default connect(homeState, {})(Home)
