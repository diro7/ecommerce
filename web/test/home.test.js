
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../src/modules/pages/Home'
 
const mockStore = configureStore([]);
 
describe('My Connected React-Redux Component', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });
 
    component = renderer.create(
      <Provider store={store}>
        <Router>
        <Home />
        </Router>);
        
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});