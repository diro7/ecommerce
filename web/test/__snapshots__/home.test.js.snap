// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Array [
  <div>
    <div
      className="jsx-856475002 "
      style={
        Object {
          "backgroundAttachment": "fixed",
          "backgroundImage": "url('http://localhost:3000/images/cover.jpg')",
          "backgroundPosition": "center top",
          "backgroundSize": "cover",
          "color": "#ffffff",
          "height": "calc(100vh - 5em)",
          "textAlign": "center",
        }
      }
    >
      <div
        className="jsx-230203545 "
      >
        <h1
          className="jsx-3315563251 "
          style={
            Object {
              "textShadow": "0px 2px 2px rgba(0, 0, 0, 0.14)",
            }
          }
        >
          Crate
        </h1>
        <h4
          className="jsx-3486686285 "
          style={
            Object {
              "marginTop": "0.5em",
              "textShadow": "0px 2px 2px rgba(0, 0, 0, 0.14)",
            }
          }
        >
          Monthly subscription of trendy clothes and accessories
        </h4>
        <a
          href="/user/signup"
          onClick={[Function]}
        >
          <button
            className="jsx-2258186958 "
            disabled={false}
            style={
              Object {
                "marginTop": "1em",
              }
            }
            type="button"
          >
            Get Started
          </button>
        </a>
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "fixed",
            "right": 0,
            "top": 0,
            "transition": "opacity 0.25s ease-in-out",
            "visibility": "visible",
            "zIndex": 9,
          }
        }
      >
        <div
          style={
            Object {
              "background": "url('/images/cover.jpg') top center",
              "backgroundSize": "cover",
              "bottom": "-40px",
              "filter": "blur(25px)",
              "left": "-40px",
              "position": "fixed",
              "right": "-40px",
              "top": "-40px",
              "zIndex": -1,
            }
          }
        />
        <div
          className="jsx-856475002 "
          style={
            Object {
              "color": "#ffffff",
              "height": "100%",
              "textAlign": "center",
            }
          }
        >
          <div
            className="jsx-230203545 "
          >
            <h1
              className="jsx-3315563251 "
              style={
                Object {
                  "textShadow": "0px 2px 2px rgba(0, 0, 0, 0.14)",
                }
              }
            >
              Bienvenido a Crate
            </h1>
            <h6
              className="jsx-948395976 "
              style={
                Object {
                  "marginTop": "0.5em",
                }
              }
            >
              Su tienda en linea de 
              <br />
               clothes and accessories
            </h6>
            <button
              className="jsx-1228826222 "
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "marginTop": "1.5em",
                }
              }
              type="button"
            >
              Next
            </button>
          </div>
          <div
            className="jsx-230203545 "
          >
            <img
              alt="collage"
              src="http://localhost:3000/images/collage.png"
              style={
                Object {
                  "width": 400,
                }
              }
              title="products collage"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  ");",
]
`;
